sound/soc/sof/intel/hda-dai-ops.c

```diff
@@ -310,6 +310,27 @@ static const struct hda_dai_widget_dma_ops hda_ipc3_dma_ops = {
 	.post_trigger = hda_ipc3_post_trigger,
 };
 
+static struct hdac_ext_stream *
+hda_dspless_get_hext_stream(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
+			    struct snd_pcm_substream *substream)
+{
+	struct hdac_stream *hstream = substream->runtime->private_data;
+
+	return stream_to_hdac_ext_stream(hstream);
+}
+
+static void hda_dspless_setup_hext_stream(struct snd_sof_dev *sdev,
+					  struct hdac_ext_stream *hext_stream,
+					  unsigned int format_val)
+{
+	hext_stream->hstream.format_val = format_val;
```

Comment:
maybe add a comment that we store the format_val for later use? Otherwise this looks useless.

Actually looking at this twice it does look useless.

Reply:
It is not useless. The maxbps (sig_bits) of the real codec is only know in the BE. In DSPless mode we need to use this information on the BE, but on there it is not available (we have dummy-codec).
It needs to be saved, otherwise the DMA will be miss configured.